src/lib/libdylink.js

```diff
@@ -48,6 +48,62 @@ var LibraryDylink = {
     registerWasmPlugin();
     `,
   $preloadedWasm: {},
+
+  $locateLibraryFromFS__deps: ['$FS', '$PATH'],
+  $locateLibraryFromFS: (filename, searchDirs) => {
+    // Find the library in the filesystem
+    // returns [true, abspath] if found, [false, null] otherwise.
+    if (!(typeof FS.findObject === 'function')) {
+      // FS may not have the findObject method if the filesystem is not
+      // used in the build. Skip the search in this case.
+      return [false, null];
+    }
+    var candidates = [];
+    if (PATH.isAbs(filename)) {
+      candidates.push(filename);
+    } else if (searchDirs) {
+      for (var i = 0; i < searchDirs.length; i++) {
+        candidates.push(PATH.join(searchDirs[i], filename));
+      }
+    } else {
+      return [false, null];
+    }
+
+    for (var i = 0; i < candidates.length; i++) {
+      var path = candidates[i];
+      if (FS.findObject(path)) {
+        return [true, path];
+      }
```

Comment:
`findObject` is semi deprecated I think. This also resolves symlinks and returns the fully resolved absolute path:
```suggestion
      try {
        var res = FS.lookupPath(path);
        return [true, res.path];
      } catch(e) {}
```

Reply:
Kind of a shame we need to inject a try/crash here though.   @hoodmane do I remember talking with you about a version of `lookupPath` that doesn't throw?

Reply:
We added `noent_okay` for mkdir / O_CREAT so that it won't throw if the __last__ entry doesn't exist. Perhaps we should make a variant that returns `null` if an `ENOENT` occurs in `lookupNode`.